professional_python_exercises_2_githubcli/api/test_github_api.py

```diff
@@ -0,0 +1,228 @@
+import unittest
+from datetime import datetime
+from http import HTTPStatus
+from unittest.mock import Mock, call, patch
+
+from .github_api import GitHubAPI
+from .user_details import UserDetails
+
+
+class TestGitHubAPI(unittest.TestCase):
+    @patch("professional_python_exercises_2_githubcli.api.github_api.Github", autospec=True)
+    def test_construction(self, mock_gh_cls: Mock):
+        token = "abc123"
+
+        mock_gh = mock_gh_cls.return_value
+
+        instance = GitHubAPI(token)
+
+        assert instance.session == mock_gh
+        mock_gh_cls.assert_called_once_with(token)
+
+    @patch("professional_python_exercises_2_githubcli.api.github_api.Github", autospec=True)
+    def test_get_authenticated_user(self, mock_gh_cls: Mock):
+        token = "abc123"
+        expected_username = "tony"
+
+        mock_gh = mock_gh_cls.return_value
+        fake_gh_user = Mock()
+        fake_gh_user.login = expected_username
```

Comment:
I don't really understand what is happening here. The Mock() calls the Github() method? But then it would not be a user? Why set the .login property?